test/build.test.js

```diff
@@ -108,7 +108,7 @@ test('[build] success', assert => {
     }
   });
 
-  assert.deepEquals(Object.keys(template.Resources), ['SpotFleetRole', 'SpotFleetFunction', 'SpotFleet']);
+  assert.deepEquals(Object.keys(template.Resources), ['SpotFleetRole', 'SpotFleetFunction', 'SpotFleetLogicalName']);
```

Comment:
You should also use the provided logical name to prefix the role and function. Otherwise, these would still overlap when I try to use 2 spot fleets.

```js
assert.deepEquals(Object.keys(template.Resources), [
  'SpotFleetLogicalNameRole', 
  'SpotFleetLogicalNameFunction', 
  'SpotFleetLogicalName'
], 'btw its a pet-peeve of mine when people dont use a message in test assertions');
```